src/EditorFeatures/Core/Implementation/IntelliSense/QuickInfo/Providers/AbstractSemanticQuickInfoProvider.cs

```diff
@@ -276,8 +276,10 @@ internal abstract partial class AbstractSemanticQuickInfoProvider : AbstractQuic
             var symbols = semanticModel.GetSymbols(token, document.Project.Solution.Workspace, bindLiteralsToUnderlyingType: true, cancellationToken: cancellationToken);
 
             var bindableParent = document.GetLanguageService<ISyntaxFactsService>().GetBindableParent(token);
-            var overloads = semanticModel.GetMemberGroup(bindableParent, cancellationToken);
-
+            var parentSymbol = semanticModel.GetSymbolInfo(bindableParent).Symbol;
+            var overloads = semanticModel.GetMemberGroup(bindableParent, cancellationToken)
+                                         .Where(s => parentSymbol == null || enclosingType.InheritsFromOrEquals(parentSymbol.ContainingType) || s.IsStatic == parentSymbol.IsStatic);
+            
```

Comment:
That's a good question,  I looked around (took more time then the fix) and I think this is the place were the completion code handles this: https://github.com/dotnet/roslyn/blob/master/src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationService.cs#L541 

The structur of the code for completion is quite different and I can't find any good help method to reuse. I also want to keep the change as small as possible.

By the way, I think I found another small bug when I tried the completion logic. 
```
public class Program
 { 
    static void Main(string[] args) 
    {
        new T().M($$);
    }
}
public class T
{
    public void M() { }
    public static void M(int i) { }
}
```
Press ctrl+space at the $$. In the list that is shown (why does it include that much?) there is the parameter from the static method 'i:'